conans/test/functional/gpg_verify_test.py

```diff
@@ -0,0 +1,241 @@
+qL7yzZr/qpkNLbp2djiZw3erAtxgRVUTMYrVF1OSkg==
+=lmU2
+-----END PGP PRIVATE KEY BLOCK-----
+"""
+
+
+def _generate_gpg_data(basename):
+    outfiles = []
+
+
+    #create text file
+    with open(basename + ".txt", "w") as f:
+        f.write(TEST_MESSAGE_CONTENTS)
+        f.write('\n')
+
+    outfiles.append(basename+".txt")
+
+    #create binary file
+    with tarfile.TarFile(basename + ".tar.gz", "w") as f:
+        f.add(basename+".txt")
+
+    outfiles.append(basename + ".tar.gz")
+
+    #write public key file
+    with open("pubkey.txt", "w") as f:
+        f.write(GPG_PUBLIC_KEY)
+
+    outfiles.append("pubkey.txt")
+
+    #import private key
+    gpgbinary = "gpg.exe" if platform.system() == "Windows" else "gpg"
+    gpghome = tempfile.mkdtemp()
+    
+    _gpg = gnupg.GPG(gpgbinary=gpgbinary, gnupghome=gpghome)
+    print(gnupg.__file__)
+    _gpg.import_keys(GPG_PRIVATE_KEY)
+    _gpg.import_keys(GPG_PUBLIC_KEY)
+
+
+    if not _gpg.list_keys(keys=FINGERPRINT):
+        raise RuntimeError("importing GPG keys failed")
+
+
+    #sign binary file detached signature
+    with open(basename + ".tar.gz", "rb") as f:
+        signed = _gpg.sign_file(f, keyid=FINGERPRINT[-8:],
+                                passphrase=PASSPHRASE, detach=True,
+                                binary=True, output=basename+".tar.gz.sig")
+
+    if not signed:
+        raise RuntimeError("failed to sign test data")
+
+    #sign binary file detached ascii signature
+    with open(basename + ".tar.gz","rb") as f:
+        signed = _gpg.sign_file(f, keyid=FINGERPRINT[-8:],
+                                passphrase=PASSPHRASE, detach=True,
+                                binary=False, output=basename+".tar.gz.asc")
+
+    outfiles.append(basename + ".tar.gz.sig")
+
+
+
+    #delete private and public keys
+    deleted_priv = _gpg.delete_keys(FINGERPRINT, secret=True)
+    if not deleted_priv:
+        raise RuntimeError("failed to delete private key: %s"
+                           % deleted_priv.status)
+
+    deleted_pub = _gpg.delete_keys(FINGERPRINT)
+    if not deleted_pub:
+        raise RuntimeError("failed to delete public key: %s "
+                           % deleted_pub.status)
+
+    return outfiles, _gpg
+
+
+
+class GPGVerifyTest(unittest.TestCase):
+    def setUp(self):
+        self._data_files, self._gpg = _generate_gpg_data("gpg_test_data")
+        self._data_file = "gpg_test_data.tar.gz"
+        self._sig_file = "gpg_test_data.tar.gz.sig"
+        self._key_file = "pubkey.txt"
+        self._ascii_sig_file = "gpg_test_data.tar.gz.asc"
+
+    def tearDown(self):
+        map(os.remove, self._data_files)
+
+    def testCorrectVerifyFile(self):
+        verify_gpg_sig(data_file=self._data_file, pubkey=GPG_PUBLIC_KEY,
+                       sig_file=self._sig_file)
+
+        #if we got to here without throwing, verification completed
+
+    def testIncorrectSignatureVerifyFile(self):
```

Comment:
Need to use ``test_incorrect_...`` pep8 names.

Also make sure your code is linted, it seems you have no checker.

Reply:
I do have a checker, but probably not set high enough...